pkg/sessions/cookie/session_store.go

```diff
@@ -0,0 +1,232 @@
+package cookie
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/pusher/oauth2_proxy/cookie"
+	"github.com/pusher/oauth2_proxy/pkg/apis/options"
+	"github.com/pusher/oauth2_proxy/pkg/apis/sessions"
+	"github.com/pusher/oauth2_proxy/pkg/cookies"
+	"github.com/pusher/oauth2_proxy/pkg/sessions/utils"
+)
+
+const (
+	// Cookies are limited to 4kb including the length of the cookie name,
+	// the cookie name can be up to 256 bytes
+	maxCookieLength = 3840
+)
+
+// Ensure CookieSessionStore implements the interface
+var _ sessions.SessionStore = &SessionStore{}
+
+// SessionStore is an implementation of the sessions.SessionStore
+// interface that stores sessions in client side cookies
+type SessionStore struct {
+	CookieCipher   *cookie.Cipher
+	CookieDomain   string
```

Comment:
Should we be sending along the `options.CookieOptions` struct? This might make `makeCookie` a bit more usable across different store implementations

Reply:
That's a great idea! If we keep the `CookieOptions` struct we can move `makeCookie` to a shared package and make it reusable, I'll do that and join it to #148